### ISSUE TITLE
Failed to connect to server (code: 1005)

### ISSUE THREAD

**Issue by yogeshwarvm** [jcrodriguez-dis/vpl-xmlrpc-jail#27]
While running the program, a blank window appears and this error occurs in console.

I tried on Chrome and Firefox both. I also have set the memory limit to 512 MB (to avoid memory error). 

In
![Screenshot from 2019-07-25 15-03-19](https://user-images.githubusercontent.com/36185596/61863775-dffea300-aeed-11e9-982d-d4ef2f2fdc1a.png)
 my /var/log/messages file, this occurs every time on execution:
**live abrt-hook-ccpp: Process 10982 (Xvnc) of user 18243 killed by SIGSEGV - dumping core**

**Comment by jcrodriguez-dis** [comment]
Dear Yogeshwar Shukla,
please, tell us what is the version of your vpl-jail-system and O.S that run it.
Your report shows that the problem is the connection with the VNC server. The log about the Xvnc killed is normal after a timeout waiting for a connection.
what is your network configuration to reach from your browser to your jail server?

Best regards,
Juan Carlos.

**Comment by ypshukla** [comment]
Hi,

I am using the vpl-jail-system version  v2.3 and running on CentOS 7
I am using https (port 8443) for jail server along with Apache web server on port 443.

Please let me know if any more detail required.

**Comment by jcrodriguez-dis** [comment]
Thanks for the information.
Sorry, but as you can see in the release notes of VPL-JAIL-SYSTEM 2.3, this is a known issue that until now has not to be resolved.

If you need running programs with GUI then you must switch to Debian or Ubuntu (maybe installing Virtualbox on your CentOS).

[VPL-JAIL-SYSTEM 2.3 release notes](https://vpl.dis.ulpgc.es/index.php/home/news/83-release-of-vpljail2-3)

Best regards,
Juan Carlos.

**Comment by ypshukla** [comment]
Hi,

Thanks. I tried on Ubuntu. But I am getting some error in logs. The log file has been attached herewith. Kindly have a look.
[syslog.txt](https://github.com/jcrodriguez-dis/vpl-xmlrpc-jail/files/3450331/syslog.txt)

**Comment by jcrodriguez-dis** [comment]
Hi,
Sorry, but the log messages are not related to vpl.

Please, follow this steps to install vpl-jail-system in VirtualBox VM:
- Install a minimum (do not install any service) Ubuntu server 18.04/16.04 on a new VM. 
- Install vpl-jail-system on the VM.
- If you are using a web server on the host machine configure the vpl-jail-system to use other ports (e.g. 8080 and 8081). Edit /etc/vpl/vpl-jail-system.conf
- Configure the VM to use NAT networking and add port forwarding for the seleced ports. (e.g. 8080=>8080 and 8081=>8081)
- Check that your host firewall (if any) allow access to selected ports.

Best regards,
Juan Carlos.

**Comment by ypshukla** [comment]
Hi Juan,

I followed the same steps. Installed Ubuntu 18.04 on Virtual Box & configured the apache, mysql & php for using the Moodle. After that I installed the vpl-jail-system & provided the port 8080 & 8443 to vpl. But as my previous log I was getting those errors. 

I made a little change in /etc/vpl/vpl-jail-system.conf file
ALLOWSUID=false to 
ALLOWSUID=true

After making this change the **fuser mount** error resolved. But still I am getting some error in logs:

[syslog.txt](https://github.com/jcrodriguez-dis/vpl-xmlrpc-jail/files/3460793/syslog.txt)

Kindly help me with the same, why I am getting these errors.

**Comment by jcrodriguez-dis** [comment]
Hi
sorry, maybe I do not explain properly the proposed solution.
My last post supposes you are using as base O.S. CentOS and you can not switch to Ubuntu or Debian. If you can switch to Ubuntu or Debian you don't need VirtualBox.
If you install vpl-jail-service on a VirtualBox machine you don't need to install mysql, apache or Moodle on your VM. I guessed that Moodle is already installed in your CentOS. Also, it is better to have both services isolated, separating vpl-jail-service from Moodle. As you are short of resources, it is better to do not install desktop software on your vpl-jail-service virtual machine.

Really I don't know the reason the logs you get. The logs do not seem to be related to VPL. If stop the vpl-jail-service you do not get that logs?

Best regards,
Juan Carlos.

**Comment by ypshukla** [comment]
Hi Juan,

Thank you so much for the explanation. I also followed the same steps. The web server & Moodle is running on my CentOS system. I have installed the Virtual Box in CentOS & have a minimal installed Ubuntu on Virtual Box. I configured the vpl jail server on Ubuntu & tested with **URL/OK**, it's working.

I know the error logs are not related to vpl but I am unable to figure it out that why I am getting these errors in log & only a blank blue screen appears when try to run the code on Moodle.

Screenshots also attached herewith.
![Screenshot from 2019-08-06 15-41-52](https://user-images.githubusercontent.com/36185596/62531824-d6b3f580-b860-11e9-9423-10fe5a58a224.png)
![Screenshot from 2019-08-06 15-41-32](https://user-images.githubusercontent.com/36185596/62531825-d6b3f580-b860-11e9-97dc-7cce458100cb.png)
